newlib/libc/tinystdio/fclose.c

```diff
@@ -41,5 +41,6 @@ fclose(FILE *f)
 		 */
```

Comment:
Is this function missing a call to __flockfile?

Reply:
@keith-packard , that's right, thank you for spotting that!

I wanted to run tests locally but did not find how to do this in readme. Could you please point me to where I can start?

Reply:
The test that I wrote uses the underlying POSIX libary for locks and threads. There's a bunch of special build flags needed to do testing with that; the `scripts/do-native-configure` shell script sets it all up nicely.
```
$mkdir build-native
$ cd build-native
$ ../scripts/do-native-configure -Dflockfile=true
$ ninja test
```